### ISSUE TITLE
Different conv2d with bias behavior between python script and test_backend.py

### ISSUE THREAD

**Issue by hwangdeyu** [onnx/tensorflow-onnx#1792]
**Describe the bug**
Different conv2d with bias behavior between python script and [test_backend.py](https://github.com/onnx/tensorflow-onnx/blob/master/tests/test_backend.py)

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 18.04):
- Tensorflow Version: 1.15.4
- Python version: Python 3.8

**To Reproduce**
- Python script
```python
import onnx
import tf2onnx
import tensorflow as tf

inputs = tf.keras.Input(shape=(64, 256, 256))
middles = tf.keras.layers.ZeroPadding2D(
    padding=(0, 4),
    data_format="channels_first",
    name="padding")(inputs)
outputs = tf.keras.layers.Conv2D(
    filters=1,
    kernel_size=(9, 9),
    strides=(1, 1),
    use_bias=True,
    data_format="channels_first",
    name="conv2d",
)(middles)
model = tf.keras.Model(inputs=inputs, outputs=outputs)

onnx_model, _ = tf2onnx.convert.from_keras(
    model=model,
    opset=13,
    large_model=False,
)
onnx.save_model(onnx_model, "conv2d.onnx")
```

- test_backend test
```python
    def test_conv2d_biasadd_rewriter(self):
        x_shape = [1, 64, 256, 256]
        x_val = make_xval(x_shape)
        def func(x):
            middles = tf.keras.layers.ZeroPadding2D(
                padding=(0, 4),
                data_format="channels_first",
                name="padding")(x)
            t = tf.keras.layers.Conv2D(
                filters=1,
                kernel_size=(9, 9),
                strides=(1, 1),
                use_bias=True,
                data_format="channels_first",
                name="conv2d",
            )(middles)
            return tf.identity(t, name=_TFOUTPUT)
        self._run_test_case(func, [_OUTPUT], {_INPUT: x_val}, rtol=1e-04, atol=1e-2, as_session=True,
                                graph_validator=lambda g: check_op_count(g, "Add", 0, disabled=False))
```

**Screenshots tf graph**
Different node.
- Python script result
![image](https://user-images.githubusercontent.com/10047193/144208530-75b0c09a-4d7a-45e3-8338-cc3c3d9476f5.png)

- test_backend test result
![image](https://user-images.githubusercontent.com/10047193/144208393-8829392f-dac7-45f3-8d6c-ccf864a971c1.png)

[graph_pb.zip](https://github.com/onnx/tensorflow-onnx/files/7632636/graph_pb.zip)

**Comment by hwangdeyu** [comment]
Seems test_backend test result is unexpectation. It probably a constfold issue.

**Comment by guschmue** [comment]
hm, on tf-2.7 this works fine. On tf-1.15 I see it (mine has a Add instead of BiasAdd).

**Comment by guschmue** [comment]
For tf-2.x we call
from_keras() -> from_function() -> tf_optimize()

But for tf-1.15 we use the old style keras session api to get the graph in _from_keras_tf1() and that code is missing to call tf_optimize().
@hwangdeyu, in _from_keras_tf1() if you call  tf_optimize() after tf_loader.freeze_session() this might be fixed.

There is some low risk that this might still not work because on tf-1.x we'd not use grappler - we'd need to test to make sure it is fine.

**Comment by hwangdeyu** [comment]
> For tf-2.x we call from_keras() -> from_function() -> tf_optimize()
> 
> But for tf-1.15 we use the old style keras session api to get the graph in _from_keras_tf1() and that code is missing to call tf_optimize(). @hwangdeyu Deyu Huang FTE, in _from_keras_tf1() if you call tf_optimize() after tf_loader.freeze_session() this might be fixed.

Yeah, I have tried to call tf_optimize() after tf_loader.freeze_session(), but it's weird, will throw below error:
```
Init node conv2d/kernel/Assign doesn't exist in graph
```

**Comment by hwangdeyu** [comment]
> For tf-2.x we call from_keras() -> from_function() -> tf_optimize()
> 
> But for tf-1.15 we use the old style keras session api to get the graph in _from_keras_tf1() and that code is missing to call tf_optimize(). @hwangdeyu Deyu Huang FTE, in _from_keras_tf1() if you call tf_optimize() after tf_loader.freeze_session() this might be fixed.

I find the file to add the `keras` test in https://github.com/onnx/tensorflow-onnx/blob/621b4b11e727d7ffa1e81ad62b6e0727ef1049a2/tests/keras2onnx_unit_tests/test_layers.py.

So I guess the different way with `from_keras()` and `_run_test_case` is by desgin?

**Comment by guschmue** [comment]
The conv2d/kernel/Assign - this might be there in the original model but should be gone after freezing. We might look a little around why we hang on to  conv2d/kernel/Assign but I think we can fix it.

The reason why _run_test_case  and from_keras look a little different:  we used to have a v1 python api that was hard to use for users. _run_test_case  is using that. A year ago we created the new api that from_keras() belongs to which mostly is a wrapper around the old api. 
In theory they do the same thing but obviously if we forget to call tf_optimize in one case that is a problem.

One could switch _run_test_case  to the new api now that it is final, but it might be more work.

from_keras_for_tf1() was added as an afterthought because initially we did not want to support keras on tf1.
It did not receive the proper testing.
